fastlane/Fastfile

```diff
@@ -78,6 +79,71 @@ lane :release do |options|
   github_release(version: version_number)
 end
 
+desc "Creates GitHub snapshot pre-release and publishes with the 'snapshot' tag"
+lane :snapshot_release do
+  stable_version   = current_version_number
+  snapshot_version = "#{stable_version}-SNAPSHOT.#{Time.now.strftime('%Y%m%d%H%M')}"
+
+  UI.message("Creating snapshot release #{snapshot_version}...")
+
+  Dir.chdir(get_root_folder) do
+    sh("npm version #{snapshot_version} --no-git-tag-version")
+    sh("npm publish --tag snapshot")
+    sh("npm version #{stable_version} --no-git-tag-version")
+  end
+
+  add_git_tag(tag: snapshot_version, message: "Snapshot #{snapshot_version}")
+  push_to_git_remote(remote: "origin", tags: true)
+
+  create_github_release(
+    version:               snapshot_version,
+    repo_name:             repo_name,
+    github_api_token:      ENV["GITHUB_TOKEN"],
+    changelog_latest_path: changelog_latest_path,
+    is_prerelease:         true
+  )
+
+  create_wpl_dependency_pr(version: snapshot_version)
+end
+
+desc "Create a PR to update the SDK dependency in the Checkout repo"
+lane :create_wpl_dependency_pr do |options|
+  version_number = options[:version]
+
+  branch_name = "purchases-js-snapshot-release-#{version_number}"
+
+  Dir.mktmpdir do |temp_dir|
+    Dir.chdir(temp_dir) do
+      sh("git", "clone", "https://#{ENV['GITHUB_TOKEN']}@github.com/RevenueCat/#{wpl_repo_name}.git", ".")
+      sh("git", "checkout", "main")
+      sh("git", "checkout", "-b", branch_name)
+
+      sh("npm", "install", "--save-exact", "@revenuecat/purchases-js@#{version_number}")
+
+      if system("git diff --quiet")
+        UI.message("No dependency update needed; skipping PR")
+        next
+      end
+
+      sh("git", "add", "package.json", "package-lock.json")
+      sh("git", "commit", "-m", "Update purchases-js to #{version_number}")
+      sh("git", "push", "--set-upstream", "origin", branch_name)
```

Comment:
This can work, but I was wondering, is it possible to trigger a workflow in the WPL repo from an API call? That way the automations for updating the version in that repo stay in that repo, and we just trigger that from here. If not, or it's more complex, I'm ok going this route for now.

Reply:
We'd have to have some infra there for this, but I'll note it down, and if I test this and works, maybe we can get it out of the door